### PR TITLE
examples: extend example to support dynamic loading

### PR DESCRIPTION
**Pull request by axic**
Follow up of #365.

The reason I've started doing these changes is two fold:
- to give a better starter example for anyone implementing evmc host (e.g. learning from Solidity's trial and error)
- to test Rust example on CI

### CODE REVIEW

File: examples/example.c [hunk 1]
```diff
@@ -4,18 +4,33 @@
...
 {
+#ifdef STATICALLY_LINKED_EXAMPLE
     struct evmc_instance* vm = evmc_create_example_vm();
+    if (!vm)
+        return -1;
     if (!evmc_is_abi_compatible(vm))
         return -1;
+#else
+    enum evmc_loader_error_code error_code;
+    struct evmc_instance *vm = evmc_load_and_configure("example-vm.so", &error_code);
```
Comment (axic) [inline examples/example.c]: @chfast can we extend the loader to check for dll extensions (`.so`, `.dylib`, `.dll`) so we can avoid declaring them in the load string?

Comment (axic) [reply examples/example.c]: Or perhaps pass the path in `argv[1]` here.

Comment (chfast) [reply examples/example.c]: `argv[1]` sounds good.

File: examples/example.c [hunk 2]
```diff
@@ -4,18 +4,36 @@
...
+#ifdef STATICALLY_LINKED_EXAMPLE
+    (void)argc;
+    (void)argv;
     struct evmc_instance* vm = evmc_create_example_vm();
+    if (!vm)
+        return -1;
     if (!evmc_is_abi_compatible(vm))
         return -1;
+#else
+    char* config_string = (argc > 1) ? argv[1] : "example-vm.so";
```
Comment (chfast) [inline examples/example.c]:
```suggestion
    const char* config_string = (argc > 1) ? argv[1] : "./example-vm.so";
```

File: examples/example.c [hunk 3]
```diff
@@ -4,18 +4,36 @@
...
+    enum evmc_loader_error_code error_code;
+    struct evmc_instance* vm = evmc_load_and_configure(config_string, &error_code);
+    if (!vm)
+    {
+        printf("Loading error: %d\n", error_code);
+        return -1;
```
Comment (chfast) [inline examples/example.c]:
```suggestion
        return error_code;
```

Comment (axic) [reply examples/example.c]: `error_code` are positive, ok with that? (should be fine as it is non-zero).

**Comment by axic** [comment]
@chfast updated

File: examples/example.c [hunk 4]
```diff
@@ -4,18 +4,36 @@
...
 
-int main()
+int main(int argc, char* argv[])
 {
+#ifdef STATICALLY_LINKED_EXAMPLE
+    (void)argc;
+    (void)argv;
     struct evmc_instance* vm = evmc_create_example_vm();
+    if (!vm)
+        return -1;
```
Comment (chfast) [inline examples/example.c]:
```suggestion
        return EVMC_LOADER_INSTANCE_CREATION_FAILURE;
```

File: examples/example.c [hunk 5]
```diff
@@ -4,18 +4,36 @@
...
     if (!evmc_is_abi_compatible(vm))
         return -1;
```
Comment (chfast) [inline examples/example.c]:
```suggestion
        return EVMC_LOADER_ABI_VERSION_MISMATCH;
```

File: examples/example.c (hunk 3 above)
Comment (chfast) [reply examples/example.c]: Success is 0. The rest is positive. I think this is ok from OS point of view. Even better than negative I believe, but not expert here.

File: examples/example.c (hunk 5 above)
Comment (axic) [reply examples/example.c]: Ok, added them.

**Pull request merged**